sympy/core/core.py

```diff
@@ -64,7 +66,7 @@ def __delattr__(self, name):
 all_classes = set()
 
 
-class BasicMeta(type):
+class BasicMeta(ABCMeta):
```

Comment:
To make abstract class this change is needed.

Reply:
I don't think we should make this kind of change to sympy just to get type checking right for this very particular case. 

If it is a good idea to include this, it should be a seperate PR, that also makes this consistent across the entire library (e.g. then the use of `NotImplementedError` should be replaced consistently).

Reply:
That’s also not what I aimed for at the beginning. That’s why I left __getitem__’s body empty previously.

Reply:
OK. I changed my mind. `NotImplementedError` may fit here, indicating that a concrete method is not implemented (instead of this abstract method).